src/sparseml/transformers/utils/helpers.py

```diff
@@ -554,3 +555,49 @@ def fetch_recipe_path(target: str):
         recipe_path = hf_hub_download(repo_id=target, filename=DEFAULT_RECIPE_NAME)
 
     return recipe_path
+
+
+def generate_mask(string: str, prompt: str, censor: str) -> str:
+    """
+    Generate a mask based on provided prompt and censor strings to obscure
+    characters in the input string.
+
+    Args:
+    :param string: The input string to be masked.
+    :param prompt: The prompt string to identify characters to keep visible.
+    :param censor: The censor string to identify characters to obscure.
+
+    Returns:
+        str: A string representing the mask where '1' indicates visible
+        characters and '0' indicates obscured characters.
+
+    """
+    mask = ["1"] * len(string)
```

Comment:
Is there a reason mask needs to be a list of chars? I think a boolean mask would be more intuitive

Reply:
The reason for not having list of bool is because we concatenate at the end. We can do list of bool and combine at the end to a string, or output list of bool, but i dont think its too far away from a programmers intuition to use "1" and "0".  I have details in docstrings and will be happy to edit if it can be impoved.